sus/num/__private/float_methods.inc

```diff
@@ -74,11 +74,19 @@ static const i32 MAX_10_EXP;
 ///
 /// This value is not constexpr because the value can differ in a constexpr
 /// evaluation context from a runtime context, leading to bugs.
-static const _self NAN;
+///
+/// We have chosen the name NaN so that we don't clash with the C++ standard
+/// library macro NaN. It's not an ideal name, but is the best we've been able
+/// to come up with so far.
+static const _self NaN;
 /// Infinity.
-static const _self INFINITY;
+///
+/// We have chosen the name INF so that we don't clash with the C++ standard
+/// library macro INFINITY. It's not an ideal name, but is the best we've been
+/// able to come up with so far.
```

Comment:
```suggestion
/// library macro INFINITY.
```
I think this one isn't so bad so I'd drop the last bit here for INF. Explaining why it differs from Rust's name is enough here I think.